aspnetcore/client-side/spa-services.md

```diff
@@ -0,0 +1,323 @@
+---
+title: Using SpaServices for Creating Single Page Applications | Microsoft Docs
+author: scottaddie
+description: Learn about the benefits of using SpaServices to build a SPA with ASP.NET Core
+keywords: ASP.NET Core, Angular, SPA, JavaScriptServices, SpaServices
+ms.author: scaddie
+manager: wpickett
+ms.date: 6/23/2017
+ms.topic: article
+ms.assetid: 4b30576b-2718-4c39-9253-a59966747893
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: client-side/spa-services
+ms.custom: H1Hack27Feb2017
+---
+# Using SpaServices for Creating Single Page Applications with ASP.NET Core
+
+By [Scott Addie](https://github.com/scottaddie)
+
+This article describes the value proposition of [SpaServices](https://github.com/aspnet/JavaScriptServices/tree/dev/src/Microsoft.AspNetCore.SpaServices) in building a Single Page Application (SPA) with ASP.NET Core.
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/client-side/spa-services/sample)
+
+<a name="using-spa-services"></a>
+
+## Using SpaServices with ASP.NET Core
+
+A SPA is a popular web application approach due to its inherent rich user experience. Integrating client-side SPA frameworks or libraries, such as [Angular](https://angular.io/) or [React](https://facebook.github.io/react/), with server-side frameworks like ASP.NET Core can be difficult. The `Microsoft.AspNetCore.SpaServices` NuGet package, or SpaServices for short, was developed to reduce friction in the integration process. It enables seamless operation between the different client and server technology stacks.
+
+<a name="what-is-spa-services"></a>
+
+## What is SpaServices?
+
+SpaServices was created as a component of the [JavaScriptServices](https://github.com/aspnet/JavaScriptServices) project, whose goal is to position ASP.NET Core as developers' preferred server-side platform for building SPAs. SpaServices is not required to develop SPAs with ASP.NET Core. SpaServices is a client framework-agnostic library, it doesn't lock you into a particular client framework.
+
+SpaServices provides useful infrastructure such as:
+* [Server-side prerendering](#server-prerendering)
+* [Webpack Dev Middleware](#webpack-dev-middleware)
+* [Hot Module Replacement](#hot-module-replacement)
+* [Routing helpers](#routing-helpers)
+
+Collectively, these infrastructure components enhance both the development workflow and the runtime experience. The components can be adopted in an  <!-- need a simple term here - piecemeal?? --> à la carte fashion.
+
+<a name="spa-services-prereqs"></a>
+
+## Prerequisites for using SpaServices
+
+To work with SpaServices, install the following:
+1. [Node.js](https://nodejs.org/) (version 6 or later) with npm
```

Comment:
Could the SpaServices package be listed here instead of repeating it in the prerequisites sections below?